### PR TITLE
Add `card_type` to documentation 

### PR DESCRIPTION
**Pull request by m-green**
### Context
We’re adding `card_type` to the information that teams can get about a payment through the API. It shows whether end users used a debit or credit card for the payment.

### Changes proposed in this pull request
- Update https://docs.payments.service.gov.uk/reporting/ and http://localhost:4567/optional_features/digital_wallets/ with documentation about `card_type`.
- Remove the 'Card types' table from https://docs.payments.service.gov.uk/api_reference/, as it's no longer accurate.
- Fix a few trailing spaces.

### Guidance to review
Please check if factually accurate.

### CODE REVIEW

File: source/api_reference/index.html.md.erb [hunk 1]
```diff
@@ -241,25 +241,6 @@ For example:
...
-| visa             | DEBIT  | Visa             |
-| visa             | CREDIT | Visa             |
-| master-card      | DEBIT  | Mastercard       |
-| master-card      | CREDIT | Mastercard       |
-| maestro          | DEBIT  | Maestro          |
-| american-express | CREDIT | American Express |
-| diners-club      | CREDIT | Diners Club      |
-| discover         | CREDIT | Discover         |
-| jcb              | CREDIT | JCB              |
-| unionpay         | CREDIT | Union Pay        |
```
Comment (alexbishop1) [inline source/api_reference/index.html.md.erb]: It turns out that when a service searches using the `card_brand` parameter, they have to use on the strings in the leftmost column (even though the `"card_brand"` property in the response will use one of the strings in the rightmost column!) so maybe it’s best not to remove this table altogether for now.

File: source/reporting/index.html.md.erb [hunk 2]
```diff
@@ -87,9 +89,11 @@ have entered when making a payment:
     "city": "London",
     "country": "GB"
   }
-},
+}
 ```
 
+The `card_type` is `null` if we did not recognise which type of card your user made the payment with.
```
Comment (alexbishop1) [inline source/reporting/index.html.md.erb]: Given the broad range of reasons why the `"card_type"` might be null, “we were unable to determine” feels more accurate to me that “we did not recognise” but I’m happy to defer to your judgement.

File: source/api_reference/index.html.md.erb (hunk 1 above)
Comment (m-green) [reply source/api_reference/index.html.md.erb]: Thanks Alex - I'll add the table back in.

File: source/reporting/index.html.md.erb (hunk 2 above)
Comment (m-green) [reply source/reporting/index.html.md.erb]: If I remember right, I think we started with something along the lines of "we were unable to determine" - but I asked for some tech writer help to tweak it for style, and "we did not recognise" was the consensus. Let's give it a go and tweak it later if users aren't sure what we mean.

File: source/api_reference/index.html.md.erb (hunk 1 above)
Comment (m-green) [reply source/api_reference/index.html.md.erb]: Added back in - https://github.com/alphagov/pay-tech-docs/pull/330/commits/afd6d75278d6986dd07fb6343712593f270b7eaa

**Pull request merged**